# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/CompositableClient.h
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 gfx/layers/client/CompositableClient.h
--- a/gfx/layers/client/CompositableClient.h	Thu Mar 21 23:14:26 2019 +0000
+++ b/gfx/layers/client/CompositableClient.h	Fri Mar 22 03:15:14 2019 +0000
@@ -16,7 +16,8 @@
 #include "mozilla/layers/CompositorTypes.h"
 #include "mozilla/layers/LayersTypes.h"    // for LayersBackend, TextureDumpMode
 #include "mozilla/layers/TextureClient.h"  // for TextureClient
-#include "nsISupportsImpl.h"               // for MOZ_COUNT_CTOR, etc
+#include "mozilla/webrender/WebRenderTypes.h"  // for RenderRoot
+#include "nsISupportsImpl.h"                   // for MOZ_COUNT_CTOR, etc
 
 namespace mozilla {
 namespace layers {
@@ -156,7 +157,8 @@
    *
    * See AutoRemoveTexture to automatically invoke this at the end of a scope.
    */
-  virtual void RemoveTexture(TextureClient* aTexture);
+  virtual void RemoveTexture(TextureClient* aTexture,
+                             const Maybe<wr::RenderRoot>& aRenderRoot);
 
   void InitIPDL(const CompositableHandle& aHandle);
 
@@ -188,8 +190,11 @@
  */
 struct AutoRemoveTexture {
   explicit AutoRemoveTexture(CompositableClient* aCompositable,
+                             wr::RenderRoot aRenderRoot,
                              TextureClient* aTexture = nullptr)
-      : mTexture(aTexture), mCompositable(aCompositable) {}
+      : mTexture(aTexture),
+        mCompositable(aCompositable),
+        mRenderRoot(aRenderRoot) {}
 
   ~AutoRemoveTexture();
 
@@ -197,6 +202,7 @@
 
  private:
   CompositableClient* mCompositable;
+  wr::RenderRoot mRenderRoot;
 };
 
 }  // namespace layers